src/mono/mono.proj

```diff
@@ -408,6 +408,8 @@
       <_MonoMinimal Condition="'$(Configuration)' == 'Release'">,debugger_agent,log_dest</_MonoMinimal>
       <_MonoMinimal Condition="'$(Configuration)' == 'Release' and '$(MonoEnableAssertMessages)' != 'true'">$(_MonoMinimal),assert_messages</_MonoMinimal>
       <_MonoMinimal Condition="'$(MonoWasmThreads)' != 'true'">$(_MonoMinimal),threads</_MonoMinimal>
+      <_AppropriateWasmExceptionFlags Condition="'$(WasmEnableExceptionHandling)' == 'false'">-fexceptions</_AppropriateWasmExceptionFlags>
+      <_AppropriateWasmExceptionFlags Condition="'$(WasmEnableExceptionHandling)' != 'false'">-fwasm-exceptions</_AppropriateWasmExceptionFlags>
```

Comment:
How does this work?  Users can set `WasmEnableExceptionHandling` to either true or false in their own projects - which gets picked up by `WasmApp.targets`.  But it can't affect how the runtime is compiled - that happens once in CI and then gets packed up into a nuget.  

So we build the runtime with whatever `WasmEnableExceptionHandling` value happens to be in effect on CI (I think right now the default is `true`, right?).  But if a user builds their own project with `false` we will still link it with a runtime built with `-fexceptions`.  Is that reasonable? I wouldn't think so... the runtime's code wouldn't be using the wasm EH instructions.

Reply:
So the problem is that (as far as i can tell) even if you don't use EH, building with -fexceptions causes the emscripten JS exception handling glue to get linked in, and if you build with -fwasm-exceptions it links in C++ exception handling glue instead. So if we build with a mix of both, we have trouble, aside from the question of "what happens if you put a try/catch in your code".

It's possible we can get away with these flags being wrong as long as the flags are correct at the point of the final link. I'm not really sure, I don't fully understand all these different parts of our build.

Reply:
I updated this, I'm still not confident in it but it seems to work.